### PR TITLE
fix(NumberKeyboard): lose the number 0 key in some configs

### PR DESCRIPTION
**Pull request by Whbbit1999**
Before submitting a pull request, please read the [contributing guide](https://vant-ui.github.io/vant/#/en-US/contribution).

在提交 pull request 之前，请阅读 [贡献指南](https://vant-ui.github.io/vant/#/zh-CN/contribution)。

### CODE REVIEW

File: packages/vant/src/number-keyboard/NumberKeyboard.tsx
Comment (chenjiahan) [inline packages/vant/src/number-keyboard/NumberKeyboard.tsx:1]: Can you add a test case for this issue?

Comment (Whbbit1999) [reply packages/vant/src/number-keyboard/NumberKeyboard.tsx:1]: I don't really know how to write a test😄

Comment (chenjiahan) [reply packages/vant/src/number-keyboard/NumberKeyboard.tsx:1]: 单测框架是 Vitest，可以参考下其他目录的单测代码哈，参考一下加起来很容易的

Comment (Whbbit1999) [reply packages/vant/src/number-keyboard/NumberKeyboard.tsx:1]: 已添加单元测试

**Comment by PeyFon** [comment]
这个月还能合并更新上去吗:sweat_smile

**Review by chenjiahan — APPROVED** [review]

**Comment by PeyFon** [comment]
不在这一版？
image

**Comment by chenjiahan** [comment]
https://github.com/youzan/vant/releases/tag/v4.9.6

**Pull request merged**